src/controller/CHIPDeviceController.cpp

```diff
@@ -2782,6 +2823,14 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         );
     }
     break;
+    case CommissioningStage::kIcdIdentification: {
+        app::AttributePathParams readPaths[1];
+        // TODO(#29382): We probably want to read "ActiveMode" attribute (to be implemented) for ICD.
+        readPaths[0] =
+            app::AttributePathParams(app::Clusters::IcdManagement::Id, app::Clusters::IcdManagement::Attributes::FeatureMap::Id);
+        SendCommissioningReadRequest(proxy, timeout, readPaths, 1);
```

Comment:
Why is this a separate read and not being combined with the possible check-for-matching-fabric read?  We should be trying to minimize the number of reads we put on the wire....

Reply:
The matching fabric bit can be renamed to make it clear that it's reading whatever else needs to be read.

Trying to do multiple async things as part of a single stage is a bit of a pain.  Doable, of course, but more complicated than having one stage per async thing, given the current setup.

Reply:
I think it is fine to rename matching fabric bit to ReadCommissioingInfo or something.

Reply:
@bzbarsky-apple @mkardous-silabs Should we update the spec that for ICD during commissioning flow, it should try to merged into the existing read operation for commissioning, and reduce the IM read roundtrip time.

Reply:
This does not need any spec updates as far as I know.

Reply:
I now moved this and read fabric info into a kReadCommissioningInfo2

ReadCommissioningInfo2 will be a no-op if no actual read is needed.

Reply:
Mark this comment as resolved, since the code has been updated.

Modifing the content of `kReadCommissioningInfo` should be a separate refactor if required.